platforms/ide/tooling-api/src/integTest/groovy/org/gradle/integtests/tooling/ToolingApiClasspathIntegrationTest.groovy

```diff
@@ -54,17 +54,37 @@ class ToolingApiClasspathIntegrationTest extends AbstractIntegrationSpec {
                     assert files.size() == 2
                     assert files.find { it.name ==~ /slf4j-api-.*\\.jar/ } != null
 
-                    // If this suddenly fails without an obvious reason, you likely have added some code
-                    // that references types that were previously eliminated from gradle-tooling-api.jar.
-                    def jarSize = files.find { it.name ==~ /gradle-tooling-api.*\\.jar/ }.size()
-                    assert jarSize < 3_600_000
-                    // If this suddenly fails you should adjust both values so the size reduction doesn't accidentally regress again.
-                    assert jarSize > 3_450_000
+                    def shadedTapiJar = files.find { it.name ==~ /gradle-tooling-api.*\\.jar/ }
+                    assert shadedTapiJar != null
+                    println "SHADED_TAPI_JAR_SIZE: " + shadedTapiJar.size() + " bytes"
                 }
             }
         """
 
-        expect:
+        when:
         succeeds("resolve")
+
+        then:
+        def actualSize = extractShadedTapiJarSize(output)
+        def actualSizeKB = (int) Math.ceil((double) actualSize / 1024)
+
+        def expectedSizeKB = 3200
```

Comment:
The refactoring here makes the change to the size (in this PR) less obvious, https://github.com/gradle/gradle/pull/34763/commits/ed270f5a1bd1bb2835dfee77f589db4a727da2e4 helps a bit, but it would be very valuable to comment what changed in the tooling api jar and why that is accepted / ok.